src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -748,4 +748,194 @@ public void errorOperationForVector() {
               is(ErrorCode.UNSUPPORTED_UPDATE_FOR_VECTOR.getMessage() + ": " + "$push"));
     }
   }
+
+  @Nested
+  @Order(1)
+  @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+  class VectorSearchExtendedCommands {
+    @Test
+    @Order(1)
+    public void setUp() {
+      insertVectorDocuments();
+    }
+
+    @Test
+    @Order(2)
+    public void findOneAndUpdate() {
+      String json =
+          """
+        {
+          "findOneAndUpdate": {
+            "sort" : {"$vector" : [0.15, 0.1, 0.1, 0.35, 0.55]},
+            "update" : {"$set" : {"status" : "active"}},
+            "options" : {"returnDocument" : "after"}
+          }
+        }
+        """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.document._id", is("3"))
+          .body("data.document.status", is("active"))
+          .body("status.matchedCount", is(1))
+          .body("status.modifiedCount", is(1))
+          .body("errors", is(nullValue()));
+    }
+
+    @Test
+    @Order(3)
+    public void updateOne() {
+      String json =
+          """
+        {
+          "updateOne": {
+            "update" : {"$set" : {"new_col": "new_val"}},
+            "sort" : {"$vector" : [0.15, 0.1, 0.1, 0.35, 0.55]}
+          }
+        }
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.matchedCount", is(1))
+          .body("status.modifiedCount", is(1))
+          .body("status.moreData", is(nullValue()))
+          .body("errors", is(nullValue()));
+      json =
+          """
+        {
+          "findOne": {
+            "filter" : {"_id" : "3"}
+          }
+        }
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.document._id", is(3))
+          .body("data.document.new_col", is("new_val"));
+    }
+
+    @Test
+    @Order(3)
+    public void findOneAndReplace() {
+      String json =
+          """
+        {
+          "findOneAndReplace": {
```

Comment:
Is it worthwhile to test `findOneAndReplace()` with no `$vector` in `replacement`?